woodwork/tests/schema/test_table_schema_init.py

```diff
@@ -379,6 +394,42 @@ def test_schema_col_descriptions_errors(sample_column_names, sample_inferred_log
         TableSchema(sample_column_names, sample_inferred_logical_types, column_descriptions=descriptions)
 
 
+def test_schema_init_with_col_origins(sample_column_names, sample_inferred_logical_types):
+    origins = {
+        'age': 'base',
+        'signup_date': 'engineered'
+    }
+    schema = TableSchema(sample_column_names, sample_inferred_logical_types, column_origins=origins)
+    for name, column in schema.columns.items():
+        assert column.origin == origins.get(name)
+
+    schema_single_origin = TableSchema(sample_column_names, sample_inferred_logical_types, column_origins='base')
+    for name, column in schema_single_origin.columns.items():
+        assert column.origin == 'base'
+
+
+def test_schema_col_origins_errors(sample_column_names, sample_inferred_logical_types):
+    err_msg = 'column_origins must be a dictionary or a string'
+    with pytest.raises(TypeError, match=err_msg):
+        TableSchema(sample_column_names, sample_inferred_logical_types, column_origins=34)
+
+    origins = {
+        'invalid_col': 'not a valid column',
+        'signup_date': 'base'
+    }
+    err_msg = re.escape("column_origins contains columns that are not present in TableSchema: ['invalid_col']")
+    with pytest.raises(ColumnNotPresentError, match=err_msg):
+        TableSchema(sample_column_names, sample_inferred_logical_types, column_origins=origins)
+
+    origins = {
+        1: 'bad key',
```

Comment:
Sorry!!  Just realized i said keys in my comment when I meant values 🤦‍♀️ The `1` should hopefully get caught since it's not a column in the dataframe (and we do actually want to allow non-string column names). I was thinking more like `{'id':1}` as a case to be checking.